tests/tx/scripts/test_sighash.py

```diff
@@ -0,0 +1,321 @@
+
+import pytest
+
+from hathor.crypto.util import decode_address
+from hathor.exception import InvalidNewTransaction
+from hathor.manager import HathorManager
+from hathor.transaction import Transaction, TxInput, TxOutput
+from hathor.transaction.exceptions import InputOutputMismatch, InvalidInputData, InvalidScriptError
+from hathor.transaction.scripts.p2pkh import P2PKH
+from hathor.transaction.scripts.sighash import InputsOutputsLimit, SighashBitmask
+from hathor.util import not_none
+from tests import unittest
+from tests.utils import add_blocks_unlock_reward, create_tokens, get_genesis_key
+
+
+class BaseSighashTest(unittest.TestCase):
+    __test__ = False
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.manager1: HathorManager = self.create_peer('testnet', unlock_wallet=True, wallet_index=True)
+        self.manager2: HathorManager = self.create_peer('testnet', unlock_wallet=True, wallet_index=True)
+
+        # 1 is Alice
+        assert self.manager1.wallet
+        self.address1_b58 = self.manager1.wallet.get_unused_address()
+        self.private_key1 = self.manager1.wallet.get_private_key(self.address1_b58)
+        self.address1 = decode_address(self.address1_b58)
+
+        # 2 is Bob
+        assert self.manager2.wallet
+        self.address2_b58 = self.manager2.wallet.get_unused_address()
+        self.address2 = decode_address(self.address2_b58)
+
+        self.genesis_private_key = get_genesis_key()
+        self.genesis_block = self.manager1.tx_storage.get_transaction(self._settings.GENESIS_BLOCK_HASH)
+
+        # Add some blocks so we can spend the genesis outputs
+        add_blocks_unlock_reward(self.manager1)
+
+    @patch('hathor.transaction.scripts.opcode.is_opcode_valid', lambda _: True)
+    def test_sighash_bitmask(self) -> None:
+        # Create a new test token
+        token_creation_tx = create_tokens(self.manager1, self.address1_b58)
+        token_uid = token_creation_tx.tokens[0]
+        token_creation_utxo = token_creation_tx.outputs[0]
+        genesis_utxo = self.genesis_block.outputs[0]
+        parents = self.manager1.get_new_tx_parents()
+
+        # Alice creates an input spending all created test tokens
+        tokens_input = TxInput(not_none(token_creation_tx.hash), 0, b'')
+
+        # Alice creates an output sending half genesis HTR to herself
+        alice_output_script = P2PKH.create_output_script(self.address1)
+        htr_output = TxOutput(int(genesis_utxo.value / 2), alice_output_script)
+
+        # Alice creates an atomic swap tx that's missing only Bob's input, with half genesis HTR
+        atomic_swap_tx = Transaction(
+            weight=1,
+            inputs=[tokens_input],
+            outputs=[htr_output],
+            parents=parents,
+            tokens=[token_uid],
+            storage=self.manager1.tx_storage,
+            timestamp=token_creation_tx.timestamp + 1
+        )
+        self.manager1.cpu_mining_service.resolve(atomic_swap_tx)
```

Comment:
Why are you resolving the tx here if it's gonna be modified just after this point?

Reply:
Because the tx is going to be verified on line 99, so it needs to be resolved.

Reply:
But you are modifying the transaction on line 90, so the hash will become invalid again. Am I missing something?